examples/platforms/nrf528xx/src/radio.c

```diff
@@ -1299,3 +1313,41 @@ otError otPlatRadioConfigureEnhAckProbing(otInstance *         aInstance,
     return OT_ERROR_NOT_IMPLEMENTED;
 }
 #endif
+
+otError otPlatRadioSetChannelMaxTransmitPower(otInstance *aInstance, uint8_t aChannel, int8_t aMaxPower)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(aChannel >= OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN && aChannel <= OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MAX,
+                    error = OT_ERROR_INVALID_ARGS);
+
+    sMaxTxPowerTable[aChannel - OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MIN] = aMaxPower;
+    if (aChannel == nrf_802154_channel_get())
+    {
+        int8_t power;
+        if (!sPowerInitialized)
+        {
+            sDefaultTxPower = aMaxPower;
```

Comment:
The function `otPlatRadioSetChannelMaxTransmitPower ` is used to set the max transmit power, why we need to initialize the variable `sDefaultTxPower` at here?

Reply:
As mentioned before, the 0-value `sDefaultPower` doesn't mean we really want to output at 0dBm but just mean we haven't assigned a value to it yet. I add an initialization check to prefer use the max power for better radio quality.

Reply:
Ack, updated the logic and remove the line of setting `sDefaultTxPower` here. See the new function `GetTransmitPowerForChannel`